examples/request_no_blocking_main_thread/main.rs

```diff
@@ -0,0 +1,80 @@
+mod pokemon_client;
+
+use glib::clone;
+use gtk::glib::once_cell::sync::Lazy;
+use gtk::{glib, Application, ApplicationWindow, Label, ListBox, PolicyType, ScrolledWindow};
+use gtk::{prelude::*, PositionType};
+use tokio::runtime::Runtime;
+
+use pokemon_client::{Pokemon, PokemonClient};
+
+const APP_ID: &str = "org.gtk_rs.pokemon.list";
+static RUNTIME: Lazy<Runtime> =
+    Lazy::new(|| Runtime::new().expect("Setting up tokio runtime needs to succeed."));
+
+fn main() -> glib::ExitCode {
+    let app = Application::builder().application_id(APP_ID).build();
+    app.connect_activate(build_ui);
+    app.run()
+}
+
+fn build_ui(app: &Application) {
+    let (sender, receiver) = async_channel::bounded::<Result<Vec<Pokemon>, reqwest::Error>>(1);
+
+    RUNTIME.spawn(clone!(@strong sender => async move {
+        let mut pokemon_client = PokemonClient::new();
+        let pokemon_vec = pokemon_client.get_pokemon_list().await;
+        sender.send(pokemon_vec).await.expect("The channel needs to be open.");
+    }));
+
+    let result_poke_vec = receiver.recv_blocking().unwrap();
```

Comment:
This doesn't look good at all, instead wait for the response on  `glib::spawn_future_local`, the app won't start until a response is received

Reply:
that would avoid you duplicating the same code below, actually this code can just be removed :)